src/librbd/io/ImageRequest.cc

```diff
@@ -197,8 +232,21 @@ void ImageReadRequest<I>::send_request() {
   for (auto &object_extent : object_extents) {
     request_count += object_extent.second.size();
   }
+  
+  utime_t ts = ceph_clock_now();
+  bool update_timestamp = (image_ctx.atime_update_interval && (static_cast<uint64_t>(ts.sec()) >= image_ctx.atime_update_interval + image_ctx.get_access_timestamp())) ? true : false; 
```

Comment:
With high queue depths, it would be possible to issue multiple timestamp updates in parallel due to racing threads.  You can either protect w/ a lock like ```ImageCtx::snap_lock``` or perhaps switch to ```std::atomic``` storage for the access/modified times so that you can do a test/set loop to check for throttling.

Reply:
I don't see the timestamps ever getting updated after they pass the filter.